commit 0f08a7cc2d76f75aa1f4bece718b9c3b923d6896 (refs/changes/62/1362/3)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-01-27T19:50:27+00:00 (2 years, 8 months ago)
    
    joyent/node-cueball#70 crash in ConnectionSet during reliability testing
    Reviewed by: Cody Mello <cody.mello@joyent.com>
    Reviewed by: David Pacheco <dap@joyent.com>

diff --git a/lib/pool-monitor.js b/lib/pool-monitor.js
index 779cf32..e6041d4 100644
--- a/lib/pool-monitor.js
+++ b/lib/pool-monitor.js
@@ -78,7 +78,7 @@ CueBallPoolMonitor.prototype.toKangOptions = function () {
 		var obj = {};
 		obj.backends = pool.p_backends;
 		obj.connections = {};
-		var ks = pool.p_keys;
+		var ks = pool.p_keys.slice();
 		Object.keys(pool.p_connections).forEach(function (k) {
 			if (ks.indexOf(k) === -1)
 				ks.push(k);
@@ -118,7 +118,7 @@ CueBallPoolMonitor.prototype.toKangOptions = function () {
 		obj.backends = cset.cs_backends;
 		obj.fsms = {};
 		obj.connections = Object.keys(cset.cs_connections);
-		var ks = cset.cs_keys;
+		var ks = cset.cs_keys.slice();
 		Object.keys(cset.cs_fsms).forEach(function (k) {
 			if (ks.indexOf(k) === -1)
 				ks.push(k);
diff --git a/package.json b/package.json
index 151ef20..66801a0 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "cueball",
-  "version": "1.3.1",
+  "version": "1.3.2",
   "description": "",
   "main": "lib/index.js",
   "dependencies": {
